presto-cache/src/main/java/com/facebook/presto/cache/LocalRangeCacheManager.java

```diff
@@ -261,17 +263,32 @@ private boolean write(FileReadRequest key, byte[] data, Path newFilePath)
             else {
                 // copy previous file's data to the new file
                 byte[] previousFileBytes = Files.readAllBytes(new File(previousCacheFile.getPath().toUri()).toPath());
```

Comment:
rather than creating a large array `previousFileBytes` by reading everything from `previousCacheFile`, read the data to `previousCacheFile`  in small chunks and gradually append to `newFilePath`.